File: sdks/aec.py
"""KhulnaSoft Engineering Knowledge OS SDK (generated). Do not hand-edit."""

from typing import Any


class Client:
    """Minimal typed client for the KhulnaSoft Engineering Knowledge OS."""

    def resources(self) -> dict:
        path = '/resources'
        return self.request("GET", path)

    def resource(self, slug: str) -> dict:
        path = '/resources/{slug}'.replace( '{slug}', slug )
        return self.request("GET", path)

    def twin(self, slug: str) -> dict:
        path = '/resources/{slug}/twin'.replace( '{slug}', slug )
        return self.request("GET", path)

    def context(self, slug: str) -> dict:
        path = '/resources/{slug}/context'.replace( '{slug}', slug )
        return self.request("GET", path)

    def recommendations(self, slug: str) -> dict:
        path = '/resources/{slug}/recommendations'.replace( '{slug}', slug )
        return self.request("GET", path)

    def graph(self) -> dict:
        path = '/graph'
        return self.request("GET", path)

    def impact(self) -> dict:
        path = '/impact'
        return self.request("GET", path)

    def events(self) -> dict:
        path = '/events'
        return self.request("GET", path)

    def analytics(self) -> dict:
        path = '/analytics'
        return self.request("GET", path)

    def release(self) -> dict:
        path = '/release'
        return self.request("GET", path)

    def install_asset(self, asset_id: str) -> dict:
        """Install a reusable marketplace asset (template, prompt pack, policy, agent)."""
        return self.request("POST", "/marketplace/install", {"id": asset_id})

    def consume_artifact(self, asset_id: str) -> dict:
        """Fetch the artifact resolved from an installed marketplace asset."""
        return self.request("GET", "/marketplace?asset=" + asset_id)

    def request(self, method, path, body=None):
        """HTTP helper (swap with an httpx/requests transport)."""
        return {"path": path, "method": method, "body": body}

File: sdks/aec.rs
//! KhulnaSoft Engineering Knowledge OS SDK (generated). Do not hand-edit.

const BASE: &str = "https://khulnasoft.github.io/api";

pub struct Client;

pub fn resources() -> String {
    format!("{}/resources", BASE)
}

pub fn resource(slug: &str) -> String {
    format!("{}/resources/{}", BASE, slug)
}

pub fn twin(slug: &str) -> String {
    format!("{}/resources/{}/twin", BASE, slug)
}

pub fn context(slug: &str) -> String {
    format!("{}/resources/{}/context", BASE, slug)
}

pub fn recommendations(slug: &str) -> String {
    format!("{}/resources/{}/recommendations", BASE, slug)
}

pub fn graph() -> String {
    format!("{}/graph", BASE)
}

pub fn impact() -> String {
    format!("{}/impact", BASE)
}

pub fn events() -> String {
    format!("{}/events", BASE)
}

pub fn analytics() -> String {
    format!("{}/analytics", BASE)
}

pub fn release() -> String {
    format!("{}/release", BASE)
}

